### PR TITLE
Add notification db subsets

### PR DESCRIPTION
**Pull request by maxov**
An attempt to get rid of the DSL in #16720 in favor of something easier to comprehend. Also adds in info generation for everything so far.

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/notify/model/event/ConnectionEvent.scala [hunk 1]
```diff
@@ -36,25 +36,23 @@ object NewConnectionInvite extends NotificationKind[NewConnectionInvite] {
...
-            title = s"${inviter.firstName} ${inviter.lastName} wants to connect with you on Kifi",
-            body = s"Enjoy ${inviter.firstName}’s keeps in your search results and message ${inviter.firstName} directly.",
-            linkText = s"Respond to ${inviter.firstName}’s invitation",
-            imageUrl = inviterImage,
-            extraJson = Some(Json.obj(
-              "friend" -> BasicUser.fromUser(inviter)
-            ))
-          )
-      }
+  override def info(events: Set[NewConnectionInvite]): UsingDbSubset[NotificationInfo] = {
```
Comment (maxov) [inline kifi-backend/modules/common/app/com/keepit/notify/model/event/ConnectionEvent.scala]: @ryanpbrewster how does this look, in terms of complexity?

Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/notify/model/event/ConnectionEvent.scala]: That looks nice. :+1:

I like the way that `UsingDbSubset` works.

**Comment by maxov** [comment]
@andrewconner 
@leogrim 
@ryanpbrewster

**Comment by maxov** [comment]
@andrewconner 
@leogrim 
@ryanpbrewster

I'm ready to merge, this look good?

**Comment by andrewconner** [comment]
Looks good to me

**Comment by ryanpbrewster** [comment]
lgtm, but I'll be the first to admit that I haven't stared at this for long enough to really catch anything non-obvious.

**Pull request merged**